src/main/java/io/cdap/plugin/gcp/spanner/sink/SpannerOutputFormat.java

```diff
@@ -231,4 +243,125 @@ public void abortTask(TaskAttemptContext taskAttemptContext) {
       }
     };
   }
+
+  private static void writeArray(String name, Schema componentSchema, Object values, Mutation.WriteBuilder builder) {
+    Schema unionSchema = componentSchema.getUnionSchema(0);
```

Comment:
similar comment here, shouldn't assume the component is a union

Reply:
done